handlers/incident.go

```diff
@@ -33,6 +36,70 @@ func apiServiceIncidentsHandler(w http.ResponseWriter, r *http.Request) {
 	returnJson(service.Incidents, w, r)
 }
 
+func apiServiceIncidentsHandlerActive(w http.ResponseWriter, r *http.Request) {
+	service, err := findService(r)
+	if err != nil {
+		sendErrorJson(err, w, r)
+		return
+	}
+	visibleIncidents := getVisibleIncidentsOfService(service)
+	returnJson(visibleIncidents, w, r)
+}
+
+func apiSubServiceIncidentsHandlerActive(w http.ResponseWriter, r *http.Request) {
+	service, err := findService(r)
+	if err != nil {
+		sendErrorJson(err, w, r)
+		return
+	}
+	subService, err := findPublicSubService(r, service)
+	if err != nil {
+		sendErrorJson(err, w, r)
+		return
+	}
+
+	visibleIncidents := getVisibleIncidentsOfService(subService)
+	returnJson(visibleIncidents, w, r)
+}
+
+func getVisibleIncidentsOfService(service *services.Service) []incidents.Incident {
+	var visibleIncidents []incidents.Incident
+	var visibleIncidentIds []int64
+	for _, incident := range service.Incidents {
+		if visibilityCheck(incident) == true {
+			incidentVar := *incident
+			reverse(incidentVar.Updates)
+			visibleIncidents = append(visibleIncidents, incidentVar)
+			visibleIncidentIds = append(visibleIncidentIds, incident.Id)
+		}
+	}
+	log.Info(fmt.Sprintf("Visible Incidents of Service %v : %v", service.Name, visibleIncidentIds))
+	return visibleIncidents
+}
+
+func reverse(incidents []*incidents.IncidentUpdate) {
+	for i, j := 0, len(incidents)-1; i < j; i, j = i+1, j-1 {
+		incidents[i], incidents[j] = incidents[j], incidents[i]
+	}
+}
+
+func visibilityCheck(incident *incidents.Incident) bool {
```

Comment:
can be renamed to checkVisibility

Reply:
or isVisible

Reply:
Can be made a method of the incident pointer also

Reply:
Done